app/models/recipe.rb

```diff
@@ -1,3 +1,13 @@
 class Recipe < ApplicationRecord
    validates_presence_of :name, :complexity, :genre
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
+
+   def ingredient_names
+      Recipe.joins(:ingredients).where(id: self.id).pluck("ingredients.name")
```

Comment:
I love to see you're using a join here (it's great practice). However, if you want to do this even more easily:
`self.ingredients.pluck(:name)`